internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	defaultIndexImage        = "quay.io/operator-framework/upstream-opm-builder:latest"
+	defaultGRPCPort          = 50051
+	defaultContainerName     = "registry-grpc"
+	defaultContainerPortName = "grpc"
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	// BundleAddMode specifies the graph update mode that defines how channel graphs are updated
+	BundleAddMode string
+
+	//nolint:lll
```

Comment:
The lint lll is for we do not need to scroll the page and make easier read the code.  
If you are using Golang you will see a line with the default size that you cannot cross. See: 

<img width="947" alt="Screenshot 2020-07-08 at 19 28 26" src="https://user-images.githubusercontent.com/7708031/86956577-368ddd00-c151-11ea-934c-f4fdf64efcb0.png">

It is fine to use the // nolint:lll to ignore the check when for example we are doing the consts for the scaffolds files. But in this case, we should respect this check.